drivers/usb/device/usb_dc_stm32.c

```diff
@@ -36,6 +37,49 @@ LOG_MODULE_REGISTER(usb_dc_stm32);
 #error "Only one interface should be enabled at a time, OTG FS or OTG HS"
 #endif
 
+/*
+ * Some STM32U5xx parts support a USB HS PHY which is closed by the HSE input clock
+ * directly.  This requires specific configuration.
+ */
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_otghs) && defined(CONFIG_SOC_SERIES_STM32U5X)
+
+#define IS_EQ_PROP(inst, prop, compare_value) IS_EQ(DT_PROP(inst, prop), compare_value)
+
+#define HAVE_CLKCFG_NODE()                                                                         \
+	DT_NODE_HAS_PROP(DT_PHANDLE_BY_IDX(DT_NODELABEL(usbotg_hs), phys, 0), clock_cfg)
+
+#define CHECK_NODE(inst, prop, value) IS_EQ_PROP(inst, prop, value)
+
+#define CHECK_CLKCFG_NODE(value)                                                                   \
+	HAVE_CLKCFG_NODE() &&                                                                      \
+		CHECK_NODE(DT_PHANDLE_BY_IDX(DT_NODELABEL(usbotg_hs), phys, 0), clock_cfg, value)
+
+#if HAVE_CLKCFG_NODE()
+/*
+ * If we have a clock-cfg node in the PHY then figure out what the clock
+ * frequency should be. Default to 16MHz if it's not present.
+ */
+#if CHECK_CLKCFG_NODE(STM32U5_OTGHS_PHY_CLKSEL_16MHZ)
+#define USB_PHY_HSE_CLOCK_SPEED SYSCFG_OTG_HS_PHY_CLK_SELECT_1
+#elif CHECK_CLKCFG_NODE(STM32U5_OTGHS_PHY_CLKSEL_19P2MHZ)
+#define USB_PHY_HSE_CLOCK_SPEED SYSCFG_OTG_HS_PHY_CLK_SELECT_2
+#elif CHECK_CLKCFG_NODE(STM32U5_OTGHS_PHY_CLKSEL_20MHZ)
+#define USB_PHY_HSE_CLOCK_SPEED SYSCFG_OTG_HS_PHY_CLK_SELECT_3
+#elif CHECK_CLKCFG_NODE(STM32U5_OTGHS_PHY_CLKSEL_24MHZ)
+#define USB_PHY_HSE_CLOCK_SPEED SYSCFG_OTG_HS_PHY_CLK_SELECT_4
+#elif CHECK_CLKCFG_NODE(STM32U5_OTGHS_PHY_CLKSEL_26MHZ)
+#define USB_PHY_HSE_CLOCK_SPEED SYSCFG_OTG_HS_PHY_CLK_SELECT_5
+#elif CHECK_CLKCFG_NODE(STM32U5_OTGHS_PHY_CLKSEL_32MHZ)
+#define USB_PHY_HSE_CLOCK_SPEED SYSCFG_OTG_HS_PHY_CLK_SELECT_6
+#else
+#error Unsupported clock_cfg value in the usbotg_hs PHY node.
+#endif
+#else /* HAVE_CLKCFG_NODE() */
+#define USB_PHY_HSE_CLOCK_SPEED SYSCFG_OTG_HS_PHY_CLK_SELECT_1
+#endif /* HAVE_CLKCFG_NODE() */
```

Comment:
Change 3:
```suggestion
#define OTG_HS_PHY_REFERENCE_CLOCK \
  _CONCAT(SYSCFG_OTG_HS_PHY_CLK_SELECT_, \
    DT_PROP(DT_PHANDLE(DT_NODELABEL(usbotg_hs), phys), clock_cfg))
```

Reply:
The motivation behind doing it this way was to ensure only a defined value is used? Why delete that safety guard?

Reply:
Because `enum` is used in the binding, if one attempts to use an invalid value in DTS, `gen_defines` ("the DTS compiler") will error out way before this file gets compiled.

Reply:
ah! ok, yeah, oops. :-) Fixed.